hnn_core/tests/conftest.py

```diff
@@ -92,7 +93,10 @@ def _run_hnn_core_fixture(backend=None, n_procs=None, n_jobs=1,
                            't_evprox_2': 20,
                            'N_trials': 2})
             tstop = 40.
-        net = jones_2009_model(params, add_drives_from_params=True)
+            legacy_mode = False
+        # Legacy mode necessary for exact dipole comparison test
+        net = jones_2009_model(params, add_drives_from_params=True,
+                               legacy_mode=legacy_mode)
```

Comment:
The other culprit was here. When `test_parallel_backend.test_mpi_failure()` was called, it didn't expect to encounter the new legacy mode deprecation warning. The easiest way around this was to just set `legacy_mode=False` when running the reduced model simulation.